elasticsearch/healthcheck.go

```diff
@@ -0,0 +1,113 @@
+package elasticsearch
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+
+	"github.com/ONSdigital/go-ns/healthcheck"
+	"github.com/ONSdigital/go-ns/rhttp"
+
+	"net/http"
+	"net/url"
+
+	"github.com/ONSdigital/go-ns/log"
+)
+
+// ensure the elasticsearchClient satisfies the Client interface.
+var _ healthcheck.Client = (*HealthCheckClient)(nil)
+
+// List of errors
+var (
+	ErrorUnexpectedStatusCode   = errors.New("unexpected status code from api")
+	ErrorParsingBody            = errors.New("error parsing cluster health response body")
+	ErrorUnhealthyClusterStatus = errors.New("error cluster health red")
+)
+
+const unhealthy = "red"
+
+// HealthCheckClient provides a healthcheck.Client implementation for health checking elasticsearch.
+type HealthCheckClient struct {
+	cli         *rhttp.Client
+	path        string
+	serviceName string
+}
+
+// ClusterHealth represents the response from the elasticsearch cluster health check
+type ClusterHealth struct {
```

Comment:
Do we need the other items in the struct? as this only checks for the status?

Reply:
I left all the items available from request just in case the reviewer came up with other things to check before saying elastic search is okay. I will now remove unless there are any suggestions?